File: frontend/src/types/logs.ts
// TypeScript interfaces for the Log Analysis feature

export type LogSeverity = 'critical' | 'high' | 'medium' | 'low'
export type LogLinkStatus = 'linked' | 'unlinked'
export type LogFindingStatus = 'open' | 'acknowledged' | 'resolved'

export interface LogFinding {
  id: string
  scan_id: string | null
  log_group_name: string | null
  error_pattern: string | null
  severity: LogSeverity | null
  occurrence_count: number
  first_seen: string | null
  last_seen: string | null
  sample_message: string | null
  resource_id: string | null
  resource_type: string | null
  service_name: string | null
  link_status: LogLinkStatus
  status: LogFindingStatus
  ai_analysis: string | null
  ai_analyzed_at: string | null
  detected_at: string | null
}

export interface LogScan {
  id: string
  account_id: string | null
  region: string | null
  log_groups_scanned: number
  findings_count: number
  time_window_hours: number
  started_at: string | null
  completed_at: string | null
  status: 'running' | 'completed' | 'failed'
  error_message: string | null
}

export interface LogFindingsPage {
  items: LogFinding[]
  total: number
  page: number
  page_size: number
}

export interface LogAnalyzeResponse {
  finding_id: string
  analysis: string
  model: string
  analyzed_at: string
}

export type LogAnalysisStreamEvent =
  | { type: 'text_delta'; text: string }
  | { type: 'tool_use'; name: string; input: Record<string, unknown> }
  | { type: 'tool_result'; name: string; summary: string }
  | { type: 'done'; analysis: string }
  | { type: 'error'; message: string }

File: src/api/web/schemas/logs.py
"""Pydantic schemas for the log analysis API."""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


# ---------------------------------------------------------------------------
# Requests
# ---------------------------------------------------------------------------

class AnalyzeRequest(BaseModel):
    model: str = Field(default="sonnet", pattern="^(haiku|sonnet|opus)$")


# ---------------------------------------------------------------------------
# Responses
# ---------------------------------------------------------------------------

class ScanSummary(BaseModel):
    id: str
    account_id: Optional[str] = None
    region: Optional[str] = None
    log_groups_scanned: int = 0
    findings_count: int = 0
    time_window_hours: int = 24
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    status: str = "running"
    error_message: Optional[str] = None


class LogFindingResponse(BaseModel):
    id: str
    scan_id: Optional[str] = None
    log_group_name: Optional[str] = None
    error_pattern: Optional[str] = None
    severity: Optional[str] = None
    occurrence_count: int = 0
    first_seen: Optional[datetime] = None
    last_seen: Optional[datetime] = None
    sample_message: Optional[str] = None
    resource_id: Optional[str] = None
    resource_type: Optional[str] = None
    service_name: Optional[str] = None
    link_status: str = "unlinked"
    status: str = "open"
    ai_analysis: Optional[str] = None
    ai_analyzed_at: Optional[datetime] = None
    detected_at: Optional[datetime] = None


class ListFindingsResponse(BaseModel):
    items: List[LogFindingResponse]
    total: int
    page: int
    page_size: int


class AnalyzeResponse(BaseModel):
    finding_id: str
    analysis: str
    model: str
    analyzed_at: datetime
